src/shell/info.c

```diff
@@ -109,20 +165,26 @@ static int shell_init_jobinfo (flux_shell_t *shell, struct shell_info *info)
             goto out;
         }
     }
-    if (lookup_job_info_get (f_info, &jobspec, &R) < 0) {
-        shell_log_error ("error fetching jobspec,R");
+    if (lookup_jobspec_get (f_info, &jobspec) < 0) {
+        shell_log_error ("error fetching jobspec");
         goto out;
     }
     if (!(info->jobspec = jobspec_parse (jobspec, &error))) {
         shell_log_error ("error parsing jobspec: %s", error.text);
         goto out;
     }
-    if (!(info->R = json_loads (R, 0, &error))) {
-        shell_log_error ("error parsing R: %s", error.text);
-        goto out;
-    }
-    if (!(info->rcalc = rcalc_create_json (info->R))) {
-        shell_log_error ("error decoding R");
+
+    /*  Get initial version of R from initial job-info watch response:
+     */
+    resource_watch_update (info);
```

Comment:
Might want to flag in the comment that this is a synchronous RPC (although I guess it's pretty obvious!)

Reply:
`resource_watch_update()` doesn't send an RPC, but I'll add 'synchronously' to that comment before it. Note that the job shell synchronously fetched R here before here as well...